### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by ridwanediallo**
### Project requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  it 'commentsCounter should be 5' do
+    expect(subject.commentsCounter).to be(5)
+  end
+
+  it 'likesCounter should be 10' do
+    expect(subject.likesCounter).to be(10)
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/post_spec.rb]: - [ ] Kindly add a test for `last_five_comments` and `update_posts_counter` methods

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  it 'postsCounter should be an integer' do
+    subject.postsCounter = 'hello'
+    expect(subject).to_not be_valid
+  end
+
+  it 'postsCounter should be 50' do
+    expect(subject.postsCounter).to be(50)
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/user_spec.rb]: - [ ] Kindly add test for `last_three_posts` method.

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    subject { Like.new }
+    before { subject.save }
+
+    it 'Test if likesCounter is getting updated' do
+      first_post.likesCounter = 2
+      subject.post = first_post
+      expect(subject.post.likesCounter).to eq 2
+    end
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/like_spec.rb]: - [ ] Kindly add test for `update_likes_counter` method

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+RSpec.describe Comment, type: :model do
+  subject { Comment.new(text: 'Everyone come over.') }
+
+  before { subject.save }
+
+  it 'text should be present' do
+    subject.text = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/comment_spec.rb]: - [ ] Kindly add test `update_comment_counter` method.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @ridwanediallo  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Highlights 💯
✔️ Your git-flow is correct. 
✔️ You set up linter files and no errors. 
✔️ Your pull request is very professional. It has a meaningful title and summary. Keep it up.
✔️ You added validations to models according to requirements.
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] Kindly add tests for custom methods. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=of%20your%20models%27-,methods,-and%20validations.)

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by Ismailco — APPROVED** [review]
### Hi @ridwanediallo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**